Cython/Compiler/Parsing.py

```diff
@@ -3121,28 +3121,66 @@ def p_cdef_extern_block(s, pos, ctx):
 def p_c_enum_definition(s, pos, ctx):
     # s.sy == ident 'enum'
     s.next()
+
+    scoped = False
+
+    if s.sy in ('class', 'struct'):
+        scoped = True
+        s.next()
+
     if s.sy == 'IDENT':
         name = s.systring
         s.next()
         cname = p_opt_cname(s)
         if cname is None and ctx.namespace is not None:
             cname = ctx.namespace + "::" + name
     else:
-        name = None
-        cname = None
+        if scoped:
+            s.error("Unnamed scoped enum not allowed")
+        else:
+            name = None
+            cname = None
+
+    if s.sy == '(':
+        s.next()
+        underlying_type = p_c_base_type(s)
+        s.expect(')')
+    else:
+        underlying_type = Nodes.CSimpleBaseTypeNode(
+            pos,
+            name="int",
+            module_path = [],
+            is_basic_c_type = 1,
```

Comment:
Not using Python `bool` values is a legacy from the time when Python did not have a `bool` type.
```suggestion
            is_basic_c_type = True,
```

Reply:
> time when Python did not have a bool type.

That's pretty awesome :)